oauthproxy.go

```diff
@@ -623,7 +630,23 @@ func (p *OAuthProxy) IsValidRedirect(redirect string) bool {
 // IsWhitelistedRequest is used to check if auth should be skipped for this request
 func (p *OAuthProxy) IsWhitelistedRequest(req *http.Request) bool {
 	isPreflightRequestAllowed := p.skipAuthPreflight && req.Method == "OPTIONS"
-	return isPreflightRequestAllowed || p.IsWhitelistedPath(req.URL.Path)
+	return isPreflightRequestAllowed || p.IsWhitelistedPath(req.URL.Path) || p.IsWhitelistedHeader(req)
+}
+
+func (p *OAuthProxy) IsWhitelistedHeader(req *http.Request) bool {
+	for _, expr := range p.compiledHeader {
+		if expr.MatchString("Host: " + req.Host) {
+			return true
+		}
```

Comment:
Surely this can be achieved via the normal header values?

Reply:
`req.Header` does not contain the `Host` header:

> For incoming requests, the Host header is promoted to the
> Request.Host field and removed from the Header map.

https://golang.org/pkg/net/http/#Request

Reply:
This makes sense, might be good to have a comment to that effect in the code as well?